src/server/script_mgr.cc

```diff
@@ -5,33 +5,67 @@
 #include "server/script_mgr.h"
 
 #include <absl/cleanup/cleanup.h>
+#include <absl/strings/ascii.h>
+#include <absl/strings/match.h>
+#include <absl/strings/numbers.h>
 #include <absl/strings/str_cat.h>
+#include <absl/strings/str_split.h>
 
+#include <string>
+
+#include "base/flags.h"
 #include "base/logging.h"
 #include "core/interpreter.h"
 #include "facade/error.h"
 #include "server/engine_shard_set.h"
 #include "server/server_state.h"
 
+ABSL_FLAG(std::string, default_lua_config, "",
+          "Configure default mode for running Lua scripts: \n - Use 'allow-undeclared-keys' to "
+          "allow accessing undeclared keys, \n - Use 'disable-atomicity' to allow "
+          "running scripts non-atomically. \nSpecify multiple values "
+          "separated by space, for example 'allow-undeclared-keys disable-atomicity' runs scripts "
+          "non-atomically and allows accessing undeclared keys");
+
 namespace dfly {
 
 using namespace std;
 using namespace facade;
 
 ScriptMgr::ScriptMgr() {
+  // Build default script config
+  std::string config = absl::GetFlag(FLAGS_default_lua_config);
+
+  static_assert(ScriptParams{}.atomic && !ScriptParams{}.undeclared_keys);
+
+  auto parts = absl::StrSplit(config, absl::ByAnyChar(",; "), absl::SkipEmpty());
+  for (auto pragma : parts) {
+    CHECK(ScriptParams::ApplyPragma(pragma, &default_params_))
+        << "Bad format of default_lua_config flag";
+  }
+}
+
+ScriptMgr::ScriptKey::ScriptKey(string_view sha) : array{} {
+  DCHECK_EQ(sha.size(), size());
+  memcpy(data(), sha.data(), size());
 }
 
 void ScriptMgr::Run(CmdArgList args, ConnectionContext* cntx) {
   string_view subcmd = ArgS(args, 0);
 
-  if (args.size() == 1 && subcmd == "HELP") {
+  if (subcmd == "HELP") {
     string_view kHelp[] = {
         "SCRIPT <subcommand> [<arg> [value] [opt] ...]",
         "Subcommands are:",
         "EXISTS <sha1> [<sha1> ...]",
         "   Return information about the existence of the scripts in the script cache.",
         "LOAD <script>",
         "   Load a script into the scripts cache without executing it.",
+        "CONFIGURE <sha> [options ...]",
+        "   The following options are possible: ",
+        "      - Use 'allow-undeclared-keys' to allow accessing undeclared keys",
+        "      - Use 'disable-atomicity' to allow running scripts non-atomically to improve "
```

Comment:
I'd better somehow unite the format description here and at the top, but the formatting is very different